adc/proto.go

```diff
@@ -69,7 +69,7 @@ func (f Feature) String() string {
 
 func (f *Feature) UnmarshalAdc(s []byte) error {
 	if len(s) != 4 {
-		return fmt.Errorf("malformed feature [%d]", len(s))
+		return nil
```

Comment:
You need to check for a specific length here, not just silently accept any length.
Also, this should be in `Features` - someone passes 0 bytes here, so it's a bug in the caller, not in this function.